src/pylexibank/dataset.py

```diff
@@ -361,6 +407,194 @@ def coverage(self, vars, glangs, c):  # pragma: no cover
+                get_badge((lsegments - lbipapyerr) / lsegments, 'BIPA'),
+                get_badge((lsegments - lsclasserr) / lsegments, 'CLTS SoundClass'),
+            ])
+        lines.extend(['## Statistics', '\n', '\n'.join(badges), ''])
+        stats_lines = [
+            '- **Varieties:** {0:,}'.format(len(totals['languages'])),
+            '- **Concepts:** {0:,}'.format(len(totals['concepts'])),
+            '- **Lexemes:** {0:,}'.format(totals['lexemes']),
+            '- **Synonymy:** %.2f' % (totals['SI']),
+            '- **Cognacy:** {0:,} cognates in {1:,} cognate sets'.format(
+                sum(v for k, v in totals['cognate_sets'].items() if v > 1),
+                sum(1 for k, v in totals['cognate_sets'].items() if v > 1)),
+            '- **Invalid lexemes:** {0:,}'.format(stats['invalid_words_count']),
+            '- **Tokens:** {0:,}'.format(sum(stats['segments'].values())),
+            '- **Segments:** {0:,} ({1} BIPA errors, {2} CTLS sound class errors, {3} CLTS modified)'
+                .format(lsegments, lbipapyerr, lsclasserr, len(stats['replacements'])),
+            '- **Inventory size (avg):** %.2f' % stats['inventory_size'],
+            ]
+        if log:
+            log.info('\n'.join(['Summary for dataset {}'.format(self.id)] + stats_lines))
+        lines.extend(stats_lines)
+
+        totals['languages'] = len(totals['languages'])
+        totals['concepts'] = len(totals['concepts'])
+        totals['cognate_sets'] = bool(1 for k, v in totals['cognate_sets'].items() if v > 1)
+        totals['sources'] = totals['sources'].get('y', 0)
+
+        bookkeeping_languoids = []
+        for lang in self.cldf['LanguageTable']:
+            gl_lang = self.glottolog.cached_languoids.get(lang.get('Glottocode'))
+            if gl_lang and gl_lang.category == 'Bookkeeping':
+                bookkeeping_languoids.append(lang)
+
+        # improvements section
+        if len(missing_lang) or len(missing_source) or len(missing_concept) or bookkeeping_languoids:
+            lines.extend(['\n## Possible Improvements:\n', ])
+
+            if len(missing_lang):
+                lines.append("- Languages missing glottocodes: %d/%d (%.2f%%)" % (
+                    len(missing_lang),
+                    totals['languages'],
+                    (len(missing_lang) / totals['languages']) * 100
+                ))
+
+            if bookkeeping_languoids:
+                lines.append(
+                    "- Languages linked to [bookkeeping languoids in Glottolog]"
+                    "(http://glottolog.org/glottolog/glottologinformation"
+                    "#bookkeepinglanguoids):")
+            for lang in bookkeeping_languoids:
+                lines.append(
+                    '  - {0} [{1}](http://glottolog.org/resource/languoid/id/{1})'.format(
+                        lang.get('Name', lang.get('ID')), lang['Glottocode']))
+            lines.append('\n')
+
+        if len(missing_source):
```

Comment:
pyflakes tells me that it's better to use:

```if missing_source: ```

...for these types of checks (as [] evals to False)

Reply:
(which is really nitpicking, I know, sorry).